File: mpyc_demo1/utils/gpt/padding.py
import numpy as np

def pad_plaintext(plaintext):
    block_size = 16  # AES block size in bytes
    padding = block_size - (len(plaintext) % block_size)
    padded_text = plaintext + chr(padding) * padding
    return padded_text

def convert_to_bytes(padded_text):
    return [ord(char) for char in padded_text]

def create_matrix(bytes_array):
    matrix = np.zeros((4, 4), dtype=np.uint8)
    for i in range(4):
        for j in range(4):
            index = i + j * 4
            if index < len(bytes_array):
                matrix[i][j] = bytes_array[index]
            else:
                matrix[i][j] = 0  # Fill with zero if no more data
    return matrix

def prepare_for_aes(plaintext):
    padded_text = pad_plaintext(plaintext)
    bytes_array = convert_to_bytes(padded_text)
    return create_matrix(bytes_array)

# Example usage:
plaintext = "Hello, World!"
aes_input = prepare_for_aes(plaintext)
print(aes_input)

File: mpyc_demo1/utils/gpt/padding.js
function padPlaintext(plaintext) {
  const blockSize = 16; // AES block size in bytes
  const padding = blockSize - (plaintext.length % blockSize);
  const paddedText = plaintext + String.fromCharCode(padding).repeat(padding);
  return paddedText;
}

function convertToBytes(paddedText) {
  const bytes = new Uint8Array(paddedText.length);
  for (let i = 0; i < paddedText.length; i++) {
    bytes[i] = paddedText.charCodeAt(i);
  }
  return bytes;
}

function createMatrix(bytes) {
  const matrix = [];
  for (let i = 0; i < 4; i++) {
    const row = [];
    for (let j = 0; j < 4; j++) {
      row.push(bytes[i + j * 4]);
    }
    matrix.push(row);
  }
  return matrix;
}

function prepareForAES(plaintext) {
  const paddedText = padPlaintext(plaintext);
  const bytes = convertToBytes(paddedText);
  return createMatrix(bytes);
}

// Example usage:
const plaintext = 'Hello, World!';
const aesInput = prepareForAES(plaintext);
console.log(aesInput);
